__tests__/app.test.js

```diff
@@ -72,23 +83,107 @@ describe("GET: /api/reviews", () => {
-        };
         reviews.forEach((review) => {
           expect(review).toMatchObject(reviewOutput);
         });
         expect(reviews).toBeSortedBy("created_at", { descending: true });
       });
   });
+  test("should respond with expected object when queried with relevant category", () => {
+    return request(app)
+      .get("/api/reviews?category=social+deduction")
+      .expect(200)
+      .then((reviews) => {
+        const reviewObj = reviews.body;
+        expect(reviewObj).toBeInstanceOf(Object);
+        expect(reviewObj.reviews.length).toBe(11);
+        reviewObj.reviews.forEach((review) => {
+          expect(review).toMatchObject(reviewOutput);
+          expect(review.category).toBe("social deduction");
+        });
+        expect(reviewObj.reviews).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test("should respond with correctly sorted object when specified", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=votes")
+      .expect(200)
+      .then((reviews) => {
+        const reviewObj = reviews.body;
+        expect(reviewObj).toBeInstanceOf(Object);
+        reviewObj.reviews.forEach((review) => {
+          expect(review).toMatchObject(reviewOutput);
+        });
+        expect(reviewObj.reviews).toBeSortedBy("votes", {
+          descending: true,
+        });
+      });
+  });
+  test("should respond with object sorted by date ascending if specified", () => {
+    return request(app)
+      .get("/api/reviews?order=asc")
+      .expect(200)
+      .then((reviews) => {
+        const reviewObj = reviews.body;
+        expect(reviewObj).toBeInstanceOf(Object);
+        reviewObj.reviews.forEach((review) => {
+          expect(review).toMatchObject(reviewOutput);
+        });
+        expect(reviewObj.reviews).toBeSortedBy("created_at", {
+          ascending: true,
+        });
+      });
+  });
+  test("should respond with correctly filtered and sorted object when specified", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity&sort_by=votes")
+      .expect(200)
+      .then((reviews) => {
+        const reviewObj = reviews.body;
+        expect(reviewObj).toBeInstanceOf(Object);
+        reviewObj.reviews.forEach((review) => {
+          expect(review).toMatchObject(reviewOutput);
+          expect(review.category).toBe("dexterity");
+        });
+        expect(reviewObj.reviews).toBeSortedBy("votes", {
+          descending: true,
+        });
+      });
+  });
+  test("should respond with 404 if category has no reviews associated", () => {
+    return request(app)
+      .get("/api/reviews?category=children's+games")
+      .expect(404)
+      .then((err) => {
+        expect(err.body.message).toBe("No reviews for this category");
+      });
+  });
+  test("should respond with 400 if queried with invalid category field", () => {
+    return request(app)
+      .get("/api/reviews?category=mushrooms")
+      .expect(400)
```

Comment:
Could argue this should be a 404 as the category of mushrooms could eventually exist - it just doesnt currently.

Reply:
Fair, will do